polygon/dex/trades/insert_hashflow.sql

```diff
@@ -0,0 +1,96 @@
+CREATE OR REPLACE FUNCTION dex.insert_hashflow(start_ts timestamptz, end_ts timestamptz=now(), start_block numeric=0, end_block numeric=9e18) RETURNS integer
+LANGUAGE plpgsql AS $function$
+DECLARE r integer;
+BEGIN
+WITH rows AS (
+    INSERT INTO dex.trades (
+        block_time,
+        token_a_symbol,
+        token_b_symbol,
+        token_a_amount,
+        token_b_amount,
+        project,
+        version,
+        category,
+        trader_a,
+        trader_b,
+        token_a_amount_raw,
+        token_b_amount_raw,
+        usd_amount,
+        token_a_address,
+        token_b_address,
+        exchange_contract_address,
+        tx_hash,
+        tx_from,
+        tx_to,
+        trace_address,
+        evt_index,
+        trade_id
+    )
+    SELECT
+        hf.block_time,
+        maker_symbol AS token_a_symbol,
+        taker_symbol AS token_b_symbol,
+        maker_token_amount AS token_a_amount,
+        taker_token_amount AS token_b_amount,
+        'hashflow' as project,
+        NULL as version,
+        'DEX' as category,
+        pool as trader_a,
+        trader as trader_b,
+        maker_token_amount * 10 ^ erc20a.decimals as token_a_amount_raw,
+        taker_token_amount * 10 ^ erc20b.decimals as token_b_amount_raw,
+        usd_amount,
+        maker_token as token_a_address,
+        taker_token as token_b_address,
+        router_contract as exchange_contract_address,
+        tx_hash,
+        tx."from" as tx_from,
+        tx."to" as tx_to,
+        NULL::integer[] as trace_address,
+        (case when compostie_index=-1 then NULL::integer else composite_index end) as evt_index, -- -1 means decoded from traces
```

Comment:
Typo on `compostie_index`, also should be `hf.composite_index` so it's clearer where it's coming from